ambry-tools/src/main/java/com.github.ambry/store/StoreCopier.java

```diff
@@ -270,6 +270,10 @@ public void close() throws IOException {
             for (Transformer transformer : transformers) {
               message = transformer.transform(message);
             }
+            if (message == null) {
+              logger.trace("{} has no converted equivalent, treating as deleted / expired", messageInfo.getStoreKey());
```

Comment:
There may not always be conversions involved (this is just a copier) - perhaps the message should read "Dropping {} because the transformers did not return a message"

Reply:
sure, changed